### PR TITLE
Fixed Trip details spacing bug, and integrated LabeledTextFields

### PR DESCRIPTION
**Pull request by rs929**
## Overview

- Fixed trip details spacing bug
- Updated EditProfileViewController to include custom textFields

## Changes Made

- Addressed https://github.com/cuappdev/scoop-ios/issues/66
- Integrated LabeledTextField to all textFields in EditProfileVC and implemented corresponding delegates
- Implemented imagePicker Views and UIPickerViews for pronouns and year
- Fixed boolean error with email/ phone button option

## Next Steps (delete if not applicable)

- Implement validation for all textFields

### CODE REVIEW

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 1]
```diff
@@ -562,22 +538,47 @@ class EditProfileViewController: UIViewController {
...
+    
+    private func setupImagePicker() {
+        imagePicker.delegate = self
+        imagePicker.allowsEditing = true
+        imagePicker.mediaTypes = ["public.image"]
+        imagePicker.sourceType = .photoLibrary
+    }
+    
+    
+
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift]: Remove these 2 lines

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 2]
```diff
@@ -562,22 +538,47 @@ class EditProfileViewController: UIViewController {
...
     // MARK: - Helper Functions
     
+    @objc private func updateProfileImage(_ sender: UITapGestureRecognizer) {
+        present(imagePicker, animated: true)
+    }
+    
+    private func convertImage (image: UIImage) -> String {
+        let base64 = image.jpegData(compressionQuality: 0.5)?.base64EncodedString() ?? ""
+        return "data:image/png;base64," + base64
+    }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift]: Can you extract this into a `UIImage+Extension` file? See below. I think that should work.

Comment (vinnie4k) [reply Scoop/Controller/Profile/EditProfileViewController.swift]:
```
func imageToB64(compression: CGFloat) -> String {
    let base64 = self.jpegData(compressionQuality: compression)?.base64EncodedString() ?? ""
    return "data:image/png;base64," + base64
}
```

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 3]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
     }
     
 }
+
+// MARK: - UITextFieldDelegate
+
+extension EditProfileViewController: UITextFieldDelegate {
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift:623]: New line below.

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 4]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+        return !(textField == classTextField || textField == pronounsTextField)
+    }
+    
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        if let onboardingTextField = textField as? OnboardingTextField {
+            if let associatedView = onboardingTextField.associatedView as? LabeledTextField {
+                associatedView.labeledTextField(isSelected: true)
+                associatedView.hidesLabel(isHidden: false)
+            }
+        }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift]: Try this. Not sure if it works though.
```suggestion
        if let onboardingTextField = textField as? OnboardingTextField,
           let associatedView = onboardingTextField.associatedView as? LabeledTextField {
                associatedView.labeledTextField(isSelected: true)
                associatedView.hidesLabel(isHidden: false)
            }
        }
```

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 5]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if let onboardingTextField = textField as? OnboardingTextField {
+            if let associatedView = onboardingTextField.associatedView as? LabeledTextField {
+                associatedView.labeledTextField(isSelected: false)
+                if textField.text?.isEmpty ?? true {
+                    associatedView.hidesLabel(isHidden: true)
+                }
+            }
+        }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift]: Same as above.

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 6]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+    }
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.endEditing(true)
+    }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift:649]: Add new line below.

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 7]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        if pickerView == pronounsPicker {
+            return pronouns.count
+        } else if pickerView == classPicker {
+            return years.count
+        }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift:690]: Nit: add a new line below to keep it consistent with your styling above.

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 8]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+        
+        profileImageView.image = image
+
+        dismiss(animated: true, completion: nil)
+    }
+    
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        dismiss(animated: true, completion: nil)
+    }
```
Comment (vinnie4k) [inline Scoop/Controller/Profile/EditProfileViewController.swift:715]: Add a new line below when using extensions.

File: Scoop/Supporting/PickerOptions.swift [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+//
+
+import Foundation
+
+class PickerOptions {
+    
+    static let pronouns = ["He/Him", "She/Her", "They/Them", "Other"]
+    static let years = ["Freshman", "Sophomore", "Junior", "Senior", "Grad Student"]
+    
+}
```
Comment (vinnie4k) [inline Scoop/Supporting/PickerOptions.swift]: Lowkey I would create a singleton. Then use `PickerOptions.scooped.pronouns` something like that. idk if the code below works but u get the idea
```suggestion
class PickerOptions {

   static let scooped = PickerOptions()
    
   let pronouns = ["He/Him", "She/Her", "They/Them", "Other"]
   let years = ["Freshman", "Sophomore", "Junior", "Senior", "Grad Student"]
    
}
```

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 10]
```diff
@@ -20,8 +20,8 @@ class AboutYouViewController: OnboardingViewController {
     private let pronounsPicker = UIPickerView()
     private let yearPicker = UIPickerView()
     
-    private let pronouns = ["He/Him", "She/Her", "They/Them", "Other"]
-    private let years = ["Freshman", "Sophomore", "Junior", "Senior", "Grad Student"]
+    private let pronouns = PickerOptions.pronouns
+    private let years = PickerOptions.years
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift]: See the very last comment.

**Review by vinnie4k — APPROVED** [review]
Just briefly went over some stuff. Normally, you would want to filter out stuff into extensions so that you can use them throughout your other code. Gonna approve because I trust you lol.

File: Scoop/Controller/Profile/EditProfileViewController.swift [hunk 11]
```diff
@@ -623,3 +624,98 @@ class EditProfileViewController: UIViewController {
...
+        return nil
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        if pickerView == pronounsPicker {
+            pronounsTextField.textField.text = pronouns[row]
+        } else if pickerView == classPicker {
+            classTextField.textField.text = years[row]
+        }
+    }
```
Comment (tiffany-pann) [inline Scoop/Controller/Profile/EditProfileViewController.swift:673]: New line below

**Review by tiffany-pann — APPROVED** [review]
Nice work! 🚙

**Review by caitlynjin — APPROVED** [review]
LGTM! 🚙

**Pull request merged**